packages/app/models/stack-order/stack-order.ts

```diff
@@ -43,3 +44,27 @@ export const totalStacked = (order: StackOrder) =>
       acc + convertedAmount(cowOrder.executedBuyAmount, order.buyToken.decimals)
     );
   }, 0) ?? 0;
+
+const stackHasRemainingFunds = (stackOrder: StackOrder) =>
+  stackRemainingFunds(stackOrder) !== 0;
+
+export const stackRemainingFunds = (stackOrder: StackOrder) => {
+  if (totalFundsUsed(stackOrder) === 0 && totalStackOrdersDone(stackOrder) > 0)
+    return 0;
+
+  return (
+    convertedAmount(stackOrder.amount, stackOrder.sellToken.decimals) -
+    totalFundsUsed(stackOrder)
+  );
+};
+
+export const stackIsComplete = (stackOrder: StackOrder) =>
+  allOrderSlotsDone(stackOrder) && !stackHasRemainingFunds(stackOrder);
+
+export const stackIsFinishedWithFunds = (stackOrder: StackOrder) =>
+  allOrderSlotsDone(stackOrder) &&
+  stackHasRemainingFunds(stackOrder) &&
+  !stackOrder.cancelledAt;
+
+export const stackRemainingFundsWithTokenText = (stackOrder: StackOrder) =>
+  `${stackRemainingFunds(stackOrder)} ${stackOrder.sellToken.symbol}`;
```

Comment:
If we declare this inside the component, there is no need to be a function. 

```suggestion
export const stackRemainingFundsWithTokenText = `${stackRemainingFunds(stackOrder)} ${stackOrder.sellToken.symbol}`;
```

This will work inside the component, we just need to call `stackRemainingFundsWithTokenText` and not `stackRemainingFundsWithTokenText(stackOrder)`

Reply:
I recommend this because I don't see anywhere else we are calling this function.

Reply:
makes sense, I'll fix it.